src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -299,24 +299,66 @@ namespace OpenLoco::World::MapGenerator
         applySurfaceStyleToMarkedTiles(heightMap, surfaceStyle, false);
     }
 
+    // 0x0046A0D8, 0x0046A227
+    static void generateTerrainInRandomAreas([[maybe_unused]] HeightMap& heightMap, uint8_t surfaceStyle, uint16_t randAmount, uint16_t baseAmount)
+    {
+        auto ecx = getGameState().rng.randNext(64) + 0x50;
+        for (; ecx > 0; ecx--)
+        {
+            // TODO: could probably simplify / replace with two randNext(lo, hi) calls
+            auto randPos = getGameState().rng.randNext();
+            auto xPos = (((randPos & 0xFFFF) * kMapColumns) >> 16) & 0xFF;
+            auto yPos = (((randPos >> 16) * kMapRows) >> 16) & 0xFF;
+
+            auto pos = World::toWorldSpace(TilePos2(xPos, yPos));
+            auto ebx = getGameState().rng.randNext(randAmount) + baseAmount;
+            for (; ebx > 0; ebx--)
+            {
+                if (validCoords(pos))
+                {
+                    auto tile = TileManager::get(pos);
+                    auto surface = tile.surface();
+                    if (surface != nullptr)
+                    {
+                        auto height = surface->baseZ() / kSmallZStep;
```

Comment:
baseZ is already small so this is going to micro z use the kMicroToSmallZStep (same value)